clap_generate/src/lib.rs

```diff
@@ -28,6 +28,8 @@ mod macros;
 
 /// Contains some popular generators
 pub mod generators;
+/// Contains supported shells for auto-completion scripts
+pub mod shell;
```

Comment:
This does not need `pub` because we have `pub use shell::Shell` later.

Reply:
I thought `Shell` should follow the `Generator` pattern. I will update the PR.